jdbc/src/main/java/org/jclouds/jdbc/strategy/JdbcStorageStrategy.java

```diff
@@ -0,0 +1,191 @@
+import org.jclouds.jdbc.repository.ContainerRepository;
+import org.jclouds.logging.Logger;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.NoResultException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * JdbcStorageStrategy implements a blob store that stores objects
+ * on a jdbc supported database. Content metadata and user attributes are stored in
+ * the database as well.
+ */
+public class JdbcStorageStrategy implements LocalStorageStrategy {
+
+   @Resource
+   protected Logger logger = Logger.NULL;
+
+   protected final Provider<BlobBuilder> blobBuilders;
+   protected final ContainerRepository containerRepository;
+   protected final JdbcContainerNameValidator jdbcContainerNameValidator;
+   protected final JdbcBlobKeyValidator jdbcBlobKeyValidator;
+
+   @Inject
+   protected JdbcStorageStrategy(Provider<BlobBuilder> blobBuilders,
+         JdbcContainerNameValidator jdbcContainerNameValidator, JdbcBlobKeyValidator jdbcBlobKeyValidator,
+         ContainerRepository containerRepository)
+         throws ClassNotFoundException, IllegalAccessException, InstantiationException, SQLException {
+      this.containerRepository = checkNotNull(containerRepository, "containerRepository");
+      this.blobBuilders = checkNotNull(blobBuilders, "blobBuilders");
+      this.jdbcContainerNameValidator = checkNotNull(jdbcContainerNameValidator,
+            "jdbcContainerNameValidator");
+      this.jdbcBlobKeyValidator = checkNotNull(jdbcBlobKeyValidator, "jdbcBlobKeyValidator");
+   }
+
+   @Override
+   @Transactional
+   public boolean containerExists(String name) {
+      jdbcContainerNameValidator.validate(name);
+      try {
+         containerRepository.findContainerByName(name);
+      } catch (NoResultException e) {
+         return false;
+      }
+      return true;
+   }
+
+   @Override
+   @Transactional
+   public Iterable<String> getAllContainerNames() {
+      List<Container> containers = containerRepository.findAllContainers();
+      ImmutableList.Builder<String> result = ImmutableList.builder();
+      for (Container c : containers) {
+         result.add(c.getName());
+      }
+      return result.build();
+   }
+
+   @Override
+   @Transactional
+   public boolean createContainerInLocation(String name, Location location,
+         CreateContainerOptions createContainerOptions) {
+      logger.debug("Creating container %s", name);
+      jdbcContainerNameValidator.validate(name);
+      containerRepository.create(Container.builder().name(name).build());
+      return true;
+   }
+
+   @Override
+   public ContainerAccess getContainerAccess(String name) {
+      return ContainerAccess.PRIVATE;
+   }
+
+   @Override
+   public void setContainerAccess(String s, ContainerAccess containerAccess) {
+      // Unsupported
```

Comment:
What difference does it make if a container is marked on the database as private/public? I can store it in the database, but it seems pretty useless to me since the container can't be shared to other database users this way like, let's say, a S3 bucket. Again, probably I am missing something :)